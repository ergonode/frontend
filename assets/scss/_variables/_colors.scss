/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
$GREEN: #00BC87;
$GREEN_LIGHT: #EBF7EF;
$BLUE: #2096F3;
$RED: #F44336;
$RED_LIGHT: #FFF0F0;
$YELLOW: #ffc108;
$GRAPHITE_DARK: #33373E;
$GRAPHITE: #5C5F65;
$GRAPHITE_LIGHT: #85878B;
$GREY_DARK: #ADAFB2;
$GREY: #D6D7D8;
$GREY_LIGHT: #EBEBEC;
$WHITESMOKE: #F5F5F5;
$WHITE: #FFFFFF;

:export {
    GREEN: $GREEN;
    GREEN_LIGHT: $GREEN_LIGHT;
    BLUE: $BLUE;
    RED: $RED;
    RED_LIGHT: $RED_LIGHT;
    YELLOW: $YELLOW;
    GRAPHITE_DARK: $GRAPHITE_DARK;
    GRAPHITE: $GRAPHITE;
    GRAPHITE_LIGHT: $GRAPHITE_LIGHT;
    GREY_DARK: $GREY_DARK;
    GREY: $GREY;
    GREY_LIGHT: $GREY_LIGHT;
    WHITESMOKE: $WHITESMOKE;
    WHITE: $WHITE;
}
