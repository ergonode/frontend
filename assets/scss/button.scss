/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
.button {
    $button: &;

    position: relative;
    display: flex;
    flex-shrink: 0;
    justify-content: center;
    align-items: center;
    border: none;
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.5, 1);
    text-transform: uppercase;
    cursor: pointer;
    outline: none;

    &:not(&--small):not(&--tiny):not(&--fab) {
        height: 40px;
        padding: 0 12px;
        font: $FONT_SEMI_BOLD_14_20;
    }

    &:disabled {
        cursor: not-allowed;
    }

    &:hover:not(:disabled):not(&--plain) {
        box-shadow: $ELEVATOR_2_DP;
    }

    &__title {
        margin: 0 4px;
        color: $WHITE;
        letter-spacing: 0.5px;
    }

    &--primary {
        background-color: $GREEN;

        &:disabled {
            background-color: $GREY;
        }
    }

    &--secondary {
        background-color: transparent;

        #{$button}__title {
            color: $GRAPHITE;
        }

        &:disabled {
            #{$button}__title {
                color: $GREY_DARK;
            }
        }

        &:hover:not(:disabled):not(#{$button}--plain) {
            background-color: $WHITE;
        }
    }

    &--small {
        height: 32px;
        padding: 0 8px;
    }

    &--tiny {
        height: 24px;
        padding: 0 4px;
    }

    &--fab {
        width: 32px;
        height: 32px;
        border-radius: 16px;
        padding: 0;
    }

    &--tiny, &--small {
        #{$button}__title {
            font: $FONT_SEMI_BOLD_12_16;
        }
    }
}
