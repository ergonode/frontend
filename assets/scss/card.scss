/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
.card-container {
    display: block;
    min-width: 0;
    padding-bottom: 10px;
    background-color: $WHITE;
    border-radius: 2px;
    text-decoration: none;
    transition: 0.3s cubic-bezier(0.25, 0.8, 0.5, 1);

    &__shadow {
        box-shadow: $ELEVATOR_12_DP;
    }

    & .card__title {
        @include setFont(medium, medium, medium, $GRAPHITE);

        display: flex;
        flex: 1 1 auto;
        flex-wrap: wrap;
        justify-content: flex-start;
        align-items: center;
        padding: 16px;

        .title__optional {
            @include setFont(medium, small, medium, $GRAPHITE);
        }

        &--right {
            justify-content: flex-end;
        }

        &--center {
            justify-content: center;
        }

        &--toolbar {
            @include setFont(medium, large, medium, $WHITE);

            background-color: $GRAPHITE_DARK;
            box-shadow: $ELEVATOR_6_DP;
        }
    }

    & .card__content {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        padding: 8px;

        &--center {
            justify-content: center;
            align-items: center;
        }
    }

    & .card__actions {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        padding: 8px;

        &--right {
            justify-content: flex-end;
        }

        &--center {
            justify-content: center;
        }
    }
}
