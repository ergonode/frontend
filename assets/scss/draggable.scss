/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
.draggable {
    background-color: $GREY_LIGHT;
    box-shadow: $ELEVATOR_HOLE;
}
