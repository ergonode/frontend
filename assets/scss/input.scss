/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
@import "_extends/input-line";

.input {
    $input: &;

    position: relative;
    display: flex;
    flex: 1;
    flex-direction: column;
    background-color: $white;

    &__content {
        position: relative;
        display: flex;
        flex: 1;
        align-items: center;

        &--small {
            flex-basis: 32px;
        }

        &--regular {
            flex-basis: 40px;
        }

        input, textarea {
            @include setFont(medium, regular, regular, $darkGraphite);

            flex: 1;
            outline: none;
            width: 100%;
            max-width: 100%;
            min-width: 0;
            border: none;
        }

        textarea {
            height: 100%;
        }
    }

    &__information-label {
        margin-top: 2px;
    }

    &.floating-label {
        margin-top: 6px;

        #{$input}__label {
            position: absolute;
            background-color: $white;
            transition: 0.3s cubic-bezier(0.25, 0.8, 0.5, 1);
            color: $lightGraphite;
            padding-left: 2px;

            &--required {
                padding-right: 8px;
            }

            &--required::after {
                position: absolute;
                color: $error;
                content: "*";
            }
        }
    }

    &.solid {
        #{$input}__content {
            border: 1px solid $grey;
            padding: 4px;
            box-sizing: border-box;

            input, textarea {
                margin-left: 8px;
            }
        }

        #{$input}__label {
            position: absolute;
            color: $lightGraphite;
            margin-left: 6px;
        }

        #{$input}__information-label {
            margin-left: 12px;
        }

        &:hover {
            border-color: $primary;
        }
    }

    &.underline {
        #{$input}__content::before {
            bottom: 0;
            width: 100%;
            border: solid $grey;
            border-width: thin 0 0;
            @extend %defaultLine;
        }

        #{$input}__content:hover::before {
            border-color: $primary;
        }
    }

    &.left-alignment {
        #{$input}__content {
            input, textarea {
                text-align: left;
            }
        }

        #{$input}__content::after {
            bottom: 0;
            right: 100%;
            height: 2px;
            background-color: $primary;
            @extend %defaultLine;
        }
    }

    &.center-alignment {
        #{$input}__content {
            input, textarea {
                text-align: center;
            }
        }

        #{$input}__content::after {
            bottom: -1px;
            width: 100%;
            border: solid $grey;
            border-width: thin 0;
            transform: scaleX(0);
            @extend %defaultLine;
        }
    }

    &--focused {
        &.left-alignment {
            #{$input}__content::after {
                right: 0;
            }
        }

        &.center-alignment, &:not(&.solid) {
            #{$input}__content::after {
                transform: scaleX(1);
            }
        }

        &:not(#{$input}--error) {
            &.floating-label {
                #{$input}__label {
                    color: $success;
                }
            }

            &.center-alignment {
                #{$input}__content::after {
                    border-color: $primary;
                }
            }

            &.solid {
                #{$input}__content {
                    border: 2px solid $success;
                    padding: 3px;
                }
            }
        }

        &.input--error {
            &.solid {
                #{$input}__content {
                    border: 2px solid $error;
                    padding: 3px;
                }
            }

            &.floating-label {
                #{$input}__label {
                    color: $error;
                }
            }
        }
    }

    &--error {
        &.left-alignment {
            #{$input}__content::after {
                background-color: $error;
            }
        }

        &.underline {
            #{$input}__content:hover::before {
                border-color: $error;
            }
        }

        &:not(#{$input}--focused) {
            &.solid {
                #{$input}__content {
                    border: 1px solid $error;
                }
            }
        }
    }

    &--disabled {
        pointer-events: none;

        #{$input}__content {
            background-color: $background;

            input, textarea {
                background-color: $background;
                opacity: 0.4;
                color: $darkGraphite;
            }
        }
    }
}
