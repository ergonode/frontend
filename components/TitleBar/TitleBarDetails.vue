/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="title-bar__details">
        <Component
            :is="headerIconComponent"
            v-if="icon && !isBreadcrumb"
            fill-color="#00bc87" />
        <Button
            v-if="isBreadcrumb"
            class="details__back-btn"
            fab
            @click.native="onClick">
            <template #prepend>
                <IconArrowPointer fill-color="#fff" />
            </template>
        </Button>
        <div class="vertical-wrapper">
            <span
                class="details__title"
                v-text="title" />
            <slot name="badge" />
        </div>
        <slot name="breadcrumb" />
    </div>
</template>

<script>
export default {
    name: 'TitleBarDetails',
    components: {
        IconArrowPointer: () => import('~/components/Icon/Arrows/IconArrowPointer'),
        Button: () => import('~/components/Buttons/Button'),
    },
    props: {
        title: {
            type: String,
            required: true,
        },
        icon: {
            type: String,
            default: null,
        },
        isBreadcrumb: {
            type: Boolean,
            default: false,
        },
    },
    computed: {
        headerIconComponent() {
            return () => import(`~/components/Icon/Menu/Icon${this.icon}`);
        },
    },
    methods: {
        onClick() {
            this.$emit('navigateback');
        },
    },
};
</script>

<style lang="scss" scoped>
    .title-bar__details {
        display: grid;
        align-items: center;
        grid-gap: 0 8px;

        .details__back-btn {
            grid-column: 1 / 2;
            grid-row: 1 / 3;
        }

        .details__title {
            @include setFont(medium, huge, large, $darkGraphite);

            margin-right: 8px;
        }

        .vertical-wrapper {
            display: flex;
            align-items: center;
            grid-column: 2 / 3;
            grid-row: 1 / 2;
            margin-left: 8px;
        }
    }
</style>
