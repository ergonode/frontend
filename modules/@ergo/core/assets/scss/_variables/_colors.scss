/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
$GREEN_DARK: #00A979;
$GREEN: #00BC87;
$GREEN_LIGHT: #EBF7EF;
$BLUE_DARK: #0048DF;
$BLUE: #2096F3;
$RED: #F44336;
$RED_LIGHT: #FFF0F0;
$YELLOW: #FFC108;
$GRAPHITE_COAL: #252832;
$GRAPHITE_DARK: #33373E;
$GRAPHITE: #5C5F65;
$GRAPHITE_LIGHT: #85878B;
$GREY_DARK: #ADAFB2;
$GREY: #D6D7D8;
$GREY_LIGHT: #EBEBEC;
$WHITESMOKE: #F5F5F5;
$WHITE: #FFFFFF;
$BLACK: #000000;
