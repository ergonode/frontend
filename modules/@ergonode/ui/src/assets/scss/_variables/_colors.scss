/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
$GREEN_DARK: #00A979;
$GREEN: #00BC87;
$GREEN_LIGHT: #EBF7EF;
$BLUE_DARK: #0048DF;
$BLUE: #2096F3;
$BLUE_LIGHT: #DFF1FF;
$RED: #F44336;
$RED_LIGHT: #FFF0F0;
$YELLOW: #FFC108;
$YELLOW_LIGHT: #FFF6DC;
$GRAPHITE_COAL: #252832;
$GRAPHITE_DARK: #33393F;
$GRAPHITE: #5C6064;
$GRAPHITE_LIGHT: #85888B;
$GREY_DARK: #ADB0B3;
$GREY: #D6D8DA;
$GREY_LIGHT: #EBEDEF;
$WHITESMOKE: #F5F6F7;
$WHITE: #FFFFFF;
$BLACK: #000000;
