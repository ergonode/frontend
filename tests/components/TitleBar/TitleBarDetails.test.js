import { mount } from '@vue/test-utils';
import TitleBarDetails from '~/components/TitleBar/TitleBarDetails';
import { GREEN, WHITE } from '~/assets/scss/_variables/_colors.scss';

describe('TitleBar/TitleBarDetails', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(TitleBarDetails, {
            propsData: {
                title: 'test',
                isBreadcrumb: true
            },
        });
    });

    it('Component is rendered', () => {
        expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('Component is named well', () => {
        expect(typeof TitleBarDetails.name).toBe('string');
        expect(TitleBarDetails.name).toEqual('TitleBarDetails');
        expect(wrapper).toMatchSnapshot();
    });

    it('WHEN title props is set THEN show text', () => {
      expect(wrapper.contains('.details__title')).toBeTruthy();
      expect(wrapper.find('.details__title').text()).toBe('test');
    });
});
