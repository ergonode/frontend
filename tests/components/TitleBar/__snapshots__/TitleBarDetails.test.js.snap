// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar/TitleBarDetails Component is named well 1`] = `
<div class="title-bar__details">
  <!----> <button class="btn--fab details__back-btn btn btn--primary btn--regular">
    <!----> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" aria-label="box" role="presentation">
      <g fill="currentColor">
        <polygon points="20 11 7.8 11 13.4 5.4 12 4 4 12 12 20 13.4 18.6 7.8 13 20 13"></polygon>
      </g>
    </svg> </button>
  <div class="vertical-wrapper"><span class="details__title font--medium-24-32">test</span> </div>
</div>
`;
